sqlconnect/internal/snowflake/config.go

```diff
@@ -73,6 +71,9 @@ func (c Config) ConnectionString() (dsn string, err error) {
 	} else if c.UseOAuth {
 		sc.Authenticator = gosnowflake.AuthTypeOAuth
 		sc.Token = c.OAuthToken
+	} else {
+		sc.Authenticator = gosnowflake.AuthTypeSnowflake
+		sc.Password = c.Password
```

Comment:
@achettyiitr Changes in sql-connect-go seems fine, but in this case real issue is config-be where we should not have password saved as well if user has switched to UseKeyPairAuth.
Context: https://rudderlabs.slack.com/archives/C02AALVFHJN/p1740767504638659?thread_ts=1740764749.861049&cid=C02AALVFHJN

Reply:
> Also is there a bug fix we need in config-be as well? When a user switches from password to keypair, password should get removed from config/vault?

This is how the config-backend works, where we merged the config [here](https://github.com/rudderlabs/rudder-config-backend/blob/develop/src/services/destination.service.ts#L586).

> Are we passing c.UseKeyPairAuth? In that case, sc.Authenticator should be set correctly? So even with password and PrivateKey both, it will try to use keypair to connect?

Yes, we are using UseKeyPairAuth. From [this](https://github.com/snowflakedb/gosnowflake/blob/master/dsn.go#L293-L297), Password is part of the dsn, debugging why even if Authenticator set correctly password can be used.